docs/content/TextInput.mdx

```diff
@@ -47,6 +47,93 @@ TextInput is a form component to add default styling to the native text input.
 </>
 ```
 
+## Text Input with visuals and loading indicators
+
+```javascript live noinline
+const WithIconAndLoadingIndicator = () => {
+  const [isLoading, setIsLoading] = React.useState(true)
+
+  const toggleLoadingState = () => {
+    setIsLoading(!isLoading)
+  }
+
+  return (
+    <>
+      <Box mb={5} display="flex" justifyContent="flex-end">
+        <button type="button" onClick={toggleLoadingState}>
+          Toggle loading state {isLoading ? 'off' : 'on'}
+        </button>
+      </Box>
+
+      <h3>No visual</h3>
+      <Box mb={2}>
+        <TextInput value="auto" isLoading={isLoading} />
+      </Box>
+      <Box mb={2}>
+        <TextInput value="leading" isLoading={isLoading} loaderPosition="leading" />
+      </Box>
+      <Box mb={5}>
+        <TextInput value="trailing" isLoading={isLoading} loaderPosition="trailing" />
+      </Box>
+
+      <h3>Leading visual</h3>
+      <Box mb={2}>
+        <TextInput leadingVisual={SearchIcon} isLoading={isLoading} value="auto" />
+      </Box>
+      <Box mb={2}>
+        <TextInput leadingVisual={SearchIcon} isLoading={isLoading} loaderPosition="leading" value="leading" />
+      </Box>
+      <Box mb={5}>
+        <TextInput leadingVisual={SearchIcon} isLoading={isLoading} loaderPosition="trailing" value="trailing" />
+      </Box>
+
+      <h3>Trailing visual</h3>
+      <Box mb={2}>
+        <TextInput trailingVisual={SearchIcon} isLoading={isLoading} value="auto" />
+      </Box>
+      <Box mb={2}>
+        <TextInput trailingVisual={SearchIcon} isLoading={isLoading} loaderPosition="leading" value="leading" />
+      </Box>
+      <Box mb={5}>
+        <TextInput trailingVisual={SearchIcon} isLoading={isLoading} loaderPosition="trailing" value="trailing" />
+      </Box>
+
+      <h3>Both visuals</h3>
+      <Box mb={2}>
+        <TextInput
+          size="small"
+          leadingVisual={SearchIcon}
+          trailingVisual={SearchIcon}
+          isLoading={isLoading}
+          value="auto"
+        />
+      </Box>
+      <Box mb={2}>
+        <TextInput
+          leadingVisual={SearchIcon}
+          trailingVisual={SearchIcon}
+          isLoading={isLoading}
+          loaderPosition="leading"
+          value="leading"
+        />
+      </Box>
+      <Box mb={2}>
+        <TextInput
+          size="large"
+          leadingVisual={SearchIcon}
+          trailingVisual={SearchIcon}
+          isLoading={isLoading}
+          loaderPosition="trailing"
+          value="trailing"
+        />
+      </Box>
+    </>
+  )
+}
+
+render(<WithIconAndLoadingIndicator />)
```

Comment:
This could be overwhelming in the docs, I think we should move all the combinations to storybook and put the recommended/common examples in the docs

Reply:
That makes sense to me 👍